server/py/services/api/crud/secrets.py

```
@@ -596,7 +598,21 @@ def revoke_secret_token(
         )
 
     @staticmethod
-    def _validate_token_name(token_name: str, seen_names: set):
+    def _validate_token_name_and_user(
+        secret_token: mlrun.common.schemas.SecretToken,
+        seen_names: set,
+        authenticated_id: str,
+    ):
+        token_name = secret_token.name
+        token_sub = Secrets._decode_offline_token(token_name, secret_token.token).get(
```

If you decode the token here, there’s no need to decode it again in `_extract_and_validate_expiration`. You can decode the token once at the beginning, extract both the sub and exp fields, and fail early before sending the request to Iguazio.

---

I've changed the function flow to perform a single jwt decoding, then all the required validations